redis/src/types.rs

```diff
@@ -1521,96 +1520,96 @@ deref_to_write_redis_args_impl! {
 /// @note: Redis cannot store empty sets so the application has to
 /// check whether the set is empty and if so, not attempt to use that
 /// result
-impl<T: ToRedisArgs + Hash + Eq, S: BuildHasher + Default> ToRedisArgs
-    for std::collections::HashSet<T, S>
-{
-    fn write_redis_args<W>(&self, out: &mut W)
-    where
-        W: ?Sized + RedisWrite,
-    {
-        ToRedisArgs::make_arg_iter_ref(self.iter(), out)
-    }
+macro_rules! impl_to_redis_args_for_set {
+    (for <$($TypeParam:ident),+> $SetType:ty, where ($($WhereClause:tt)+) ) => {
+        impl< $($TypeParam),+ > ToRedisArgs for $SetType
+        where
+            $($WhereClause)+
+        {
+            fn write_redis_args<W>(&self, out: &mut W)
+            where
+                W: ?Sized + RedisWrite,
+            {
+                ToRedisArgs::make_arg_iter_ref(self.iter(), out)
+            }
 
-    fn num_of_args(&self) -> usize {
-        self.len()
-    }
+            fn num_of_args(&self) -> usize {
+                self.len()
+            }
+        }
+    };
 }
 
-/// @note: Redis cannot store empty sets so the application has to
-/// check whether the set is empty and if so, not attempt to use that
-/// result
-#[cfg(feature = "ahash")]
-impl<T: ToRedisArgs + Hash + Eq, S: BuildHasher + Default> ToRedisArgs for ahash::AHashSet<T, S> {
-    fn write_redis_args<W>(&self, out: &mut W)
-    where
-        W: ?Sized + RedisWrite,
-    {
-        ToRedisArgs::make_arg_iter_ref(self.iter(), out)
-    }
+impl_to_redis_args_for_set!(
+    for <T, S> std::collections::HashSet<T, S>,
+    where (T: ToRedisArgs + Hash + Eq, S: BuildHasher + Default)
+);
 
-    fn num_of_args(&self) -> usize {
-        self.len()
-    }
-}
+impl_to_redis_args_for_set!(
+    for <T> std::collections::BTreeSet<T>,
+    where (T: ToRedisArgs + Hash + Eq + Ord)
+);
 
-/// @note: Redis cannot store empty sets so the application has to
-/// check whether the set is empty and if so, not attempt to use that
-/// result
-impl<T: ToRedisArgs + Hash + Eq + Ord> ToRedisArgs for BTreeSet<T> {
-    fn write_redis_args<W>(&self, out: &mut W)
-    where
-        W: ?Sized + RedisWrite,
-    {
-        ToRedisArgs::make_arg_iter_ref(self.iter(), out)
-    }
+#[cfg(feature = "hashbrown")]
+impl_to_redis_args_for_set!(
+    for <T, S> hashbrown::HashSet<T, S>,
+    where (T: ToRedisArgs + Hash + Eq, S: BuildHasher + Default)
+);
 
-    fn num_of_args(&self) -> usize {
-        self.len()
-    }
-}
+#[cfg(feature = "ahash")]
+impl_to_redis_args_for_set!(
+    for <T, S> ahash::AHashSet<T, S>,
+    where (T: ToRedisArgs + Hash + Eq, S: BuildHasher + Default)
+);
 
-/// this flattens BTreeMap into something that goes well with HMSET
 /// @note: Redis cannot store empty sets so the application has to
```

Comment:
This can't be added to the macro. it's now here:
```rust
impl_to_redis_args_for_map!(
    /// this flattens BTreeMap into something that goes well with HMSET
    for <K, V> std::collections::BTreeMap<K, V>,
    where (K: ToRedisArgs + Hash + Eq + Ord, V: ToRedisArgs)
);
```